evaluation/evaluate.py

```diff
@@ -30,12 +31,17 @@ def send_question_to_target(question: str, url: str, parameters: dict = {}, rais
         "messages": [{"content": question, "role": "user"}],
         "context": parameters,
     }
+    
     try:
-        print(url)
         r = requests.post(url, headers=headers, json=body)
         r.encoding = "utf-8"
         latency = r.elapsed.total_seconds()
-
+                
+        if r.status_code.__str__()[0] != '2':
+            raise ConnectionError(
+                f"Request to target {url} failed with status code {r.status_code}: {r.reason} \n"
+            )
```

Comment:
What is that? Note btw that we never call under methods directly, instead you would do `str(r.status_code)`. But what are you checking for exactly?

Reply:
> we never call under methods directly, instead you would do str(r.status_code). 

yes you right, I didn't notice that, I'll change that to `str(r.status_code)`. 

> But what are you checking for exactly?

This code will check if the response to the request is returned with a success status code before parsing JSON. Because otherwise, even when the request to API returned with an error, this function will still process forward and attempt to parse the response body as JSON anyway, and got a "Value Error" that hints the error is occurred when parsing JSON, which is misleading.

And I also set the default value of the parameter `raise_error` to true. Because I think in most cases we want to explicitly throw the error instead of hiding it in result file values when meeting such kind of severe exception.

Reply:
But what does this check mean exactly?

```python
str(r.status_code)[0] != "2"
```

Reply:
Oh wow, I think I see it, you're checking whether the status code is `2xx`.

Reply:
Mate, this is absolutely horrendous :rofl:

Reply:
Yes that's right